openreview/tools.py

```diff
@@ -1,29 +1,29 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
 import openreview
 import re
 import datetime
 
 super_user_id = 'OpenReview.net'
 
-def get_profile(client, value):
-    profile = None
-    try:
-        profile = client.get_profile(value)
-    except openreview.OpenReviewException as e:
-        # throw an error if it is something other than "not found"
-        if e[0][0] != 'Profile not found':
-            print "OpenReviewException: {0}".format(e)
-            return e
-    return profile
-
 def create_profile(client, email, first, last, middle = None, allow_duplicates = False):
 
+    '''
+    Given email, first name, last name, and middle name (optional), creates and returns
+    a user profile.
+
+    If a profile with the same name exists, and allow_duplicates is False, an exception is raised.
+
+    If a profile with the same name exists and allow_duplicates is True, a profile is created with
+    the next largest number (e.g. if ~Michael_Spector1 exists, ~Michael_Spector2 will be created)
+    '''
+
     profile = get_profile(client, email)
 
     if not profile:
         response = client.get_tildeusername(first, last, middle)
-        tilde_id = response['username'].encode('utf-8')
```

Comment:
Can we get rid of the encode() now? It seems it's there for a reason.

Reply:
I've imported `unicode_literals` at the top of this file, which should solve this problem. It makes it so that this code is compatible with either python 2 or 3. 

A slightly more detailed but still brief explanation, based on my understanding: python 2 will interpret `response['username']` as a byte string, not a unicode string (byte strings are just sequences of bytes, whereas unicode is a translation of bytes to characters. This allows unicode to represent characters used by some non-english names.) Since we want `tilde_id` to account for foreign characters, we have to "encode" it to unicode (utf-8 is a type of unicode) before assigning it to the variable.

In python 3, strings are interpreted as unicode by default, so explicit encoding isn't needed. By calling `from __future__ import unicode_literals`, we're telling the python 2 interpreter to treat strings as unicode by default, as python 3 does.